.lighthouse/jenkins-x/lint-pipelines.yaml

```diff
@@ -0,0 +1,51 @@
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  creationTimestamp: null
+  name: lint-pipelines
+spec:
+  pipelineSpec:
+    tasks:
+    - name: from-build-pack
+      resources: {}
+      taskSpec:
+        metadata: {}
+        stepTemplate:
+          image: uses:jenkins-x/jx3-pipeline-catalog/tasks/go/pullrequest.yaml@versionStream
+          name: ""
+          resources:
+            requests:
+              cpu: 200m
+              memory: 256Mi
+          workingDir: /workspace/source
+        steps:
+        - image: uses:jenkins-x/jx3-pipeline-catalog/tasks/git-clone/git-clone-pr.yaml@versionStream
+          name: ""
+          resources: {}
+        - name: jx-variables
+          resources: {}
+        - name: promote-jx-preview:lint-pipelines
+          resources: {}
+          script: |
+            #!/usr/bin/env sh
+            . .jx/variables.sh
+
+            echo 'linting pipelines in the packs and .lighthouse folder:'
+            jx pipeline lint -r --out=pipelines-packs.tap
+            echo ""
+            cat pipelines-packs.tap
+
+            echo 'linting pipelines in the tasks folder:'
+            jx pipeline lint -a -d tasks --out=pipelines-tasks.tap
```

Comment:
because of the `--out=pipelines-packs.tap` arg, `jx pipeline lint -r --out=pipelines-packs.tap` always seems to return a 0 exit code. my internal jx3-pipeline-catalog seems to fail the pipeline when the linter fails. i think this is because `jx-tap run` returns a non-0 exit code when there is a failure. i can't verify this because there is no obvious way to test locally

```
jx tap run
processing tap pipelines-packs.tap
error: failed to process tap files in dir .: failed to validate pull request options: failed to validate repository options: failed to discover the repository details: failed to discover git URL in dir . you could try pass the git URL as an argument
```

```
jx tap run --help
Processes the .tap files generated by a pipeline

Usage:
  jx-tap run [flags]

Examples:
  # processes the generated tap files
  jx tap run%!(EXTRA string=jx tap)

Flags:
  -b, --batch-mode                   Runs in batch mode without prompting for user input
  -d, --dir string                   The directory to scan for *.tap files (default ".")
  -h, --help                         help for run
      --html                         generates HTML rather than commenting on the Pull Request
      --log-level string             Sets the logging level. If not specified defaults to $JX_LOG_LEVEL
  -n, --namespace string             The namespace. Defaults to the current namespace
      --verbose                      Enables verbose output. The environment variable JX_LOG_LEVEL has precedence over this flag and allows setting the logging level to any value of: panic, fatal, error, warn, info, debug, trace
      --write-log-timeout duration   The timeout for writing pipeline logs to the bucket (default 30m0s)
```